### ISSUE TITLE
Unidentified Game (No support for BF4-P UK Region)

### ISSUE THREAD

**Issue by Realinho** [rig0/BF-Launcher#3]
Greetings,

First of all, thank you for this. I've been looking for a way to play BF4 using its new launcher (which I quite like) and via Steam's in-home streaming, and just stumbled upon your launcher which seems to do exactly what I'm looking for.

After following your instructions carefully, I can't seem to be able to launch the game. After running BF-Launcher, Origin quits, starts and presents me with a message stating that activation is required for an unidentified game: 
<img width="1587" alt="screen shot 2017-05-22 at 17 12 17" src="https://cloud.githubusercontent.com/assets/22327048/26318763/1d2af5e0-3f14-11e7-98e5-07b41b0ab0d2.png">
I have the premium edition and I'm using the BF4-P line on the target box. I also tried the other lines, just to troubleshoot it, and the same thing happens.
I would greatly appreciate any help.

Thanks!

**Comment by rig0** [comment]
Hi,

I apologize for the late response. I've been really busy with other things lately. Initially the launcher only worked with 1 version of BF-4 and it was the one provided with the origin vault. Another programmer brought it to my attention that there are multiple Origin ID's (ex. OFB-EAST:109552316) for the different versions of BF4 and wrote in a few more lines of code to support them. It seems the Origin ID's might not be accurately lining up to the battlefield versions. Have you tried any of the other BF4 commands at all? I will look into the different BF4 id's myself. 

PS. It would help if you tracked down the Origin ID of the version you are using. 
You can do so by:
1. Downloading `Process Explorer` and opening it up
2. Right-click the `bf4.exe` and choose `Properties`
3. Click on the `Environment` tab 
4. Scroll down to the Variable `OID_TELEMETRY_PRODID`
5. The `Value` is the Origin ID for your version

This is optional but it would help a lot.
Thanks

**Comment by rig0** [comment]
I have tested the Origin Id for Battlefield 4 Premium and it works. I also had 2 friends try it out and it worked for them as well. Make sure that 1.) You have ran BF4 at least once after installing it and that 2.) you have all the DLC's required for Premium installed.

**Comment by Realinho** [comment]
I'm the one who's sorry for the late response: I made sure to subscribe to the thread, but didn't get any reply notifications; found out your reply right now by chance.

I was using 1.1.0 at the time and I just redownloaded it and tried again, just in case, to the same result. BF4 was ran before trying to launch BF-Launcher and I can confirm that I have all of the DLC installed.
As per your instructions, the version's value is: 1503402442.
Could it be, by any chance, anything region related? All of my Origin games were purchased from Origin's UK store.

**Comment by rig0** [comment]
Yes, you are right it is `OIG_TELEMETRY_PRODID` and not `OID_TELEMETRY_PRODID`. The value you gave me seems to be the value for `OIG_TELEMETRY_TS` though. The value should be something along the lines of `OFB-EAST:109552316`. Region could be a factor I will look into that. If you could please confirm the correct Origin ID it would be of great use.

**Comment by Realinho** [comment]
Oops, I'm sorry: I just retraced my steps and realised I gave you wrong information. Please disregard the `1503402442` value I gave you in my previous message. The requested value for `OIG_TELEMETRY_PRODID` is `OFB-EAST:109552312`.

I really appreciate you taking the time to troubleshoot this: your tool is a lifesaver to anyone who plays the game streaming it to another computer.

**Comment by rig0** [comment]
I somehow missed your reply. `OFB-EAST:109552312` is not coded into the program as it is for a different region and I only tested US versions. I did not expect them to have different Origin ID's for different regions. I don't have a way of making sure all regions and versions are covered but I could implement a line of code for your region and version. My understanding is that you have the Premium version--which region are you in? This way I can properly document it on the ReadMe file.

**Comment by Realinho** [comment]
That's correct: I have the premium version which was purchased, at the time, from Origin's UK store.

**Comment by rig0** [comment]
I have added support for your version. I will be releasing another program within the next day that will make it possible to launch most of your origin games from steam. If you have more origin games you want to launch from steam, stay tuned!

**Comment by rig0** [comment]
I will actually leave this open so you can confirm it works with your version.

**Comment by Realinho** [comment]
Many thanks for your hard work! I downloaded the updated version, set it up and can now confirm it works perfectly on the host. Opening it through Steam correctly launches BF4's new main menu and Steam finally "sees me" as playing Battlefield 4. For this alone, I'm already thankful.

However, I still can't seem to be able to play the game via In-Home Streaming. At this point, I'm convinced this may be a problem with Steam's IHS itself. After launching the game on the client, BF4's menu does appear on the host, but only a black screen with a cursor appears on the client. Moving the cursor around on the client doesn't make the host's cursor move. Also, after joining a server on the host, I can also hear audio from the client, but can only see the same black screen with the cursor.
I've been trying to troubleshoot this for the past hours, and will continue to do so until I run out of ideas. Once I do so, I'll get back to you. I'll also record a video to show you exactly what's happening.

PS: I'm definitively looking forward for other Origin games support as well, considering SimCity is another game I often play, but doesn't work with IHS.

**Comment by Realinho** [comment]
Ok rig0, I've spent the last few days trying to troubleshoot what's happening unsuccessfully and just uploaded a private video showing you what happens: https://www.youtube.com/watch?v=_zgvaLqS5mk&feature=youtu.be

I tried tinkering with the game's visual settings (different resolutions, fullscreen/ windowed), Steam's IHS settings on both host and client, installing different GPU drivers and trying two different systems as clients, to no avail. Most of the time, I open the launcher through the client, IHS starts and then quits after a few seconds with no message (I figure it's timing out). Some other times, less frequently, IHS starts but only the aforementioned black screen with a mouse cursor shows, as seen on the video. On all occasions, BF-Launcher correctly starts and launches BF4 on the host.
 I can't figure out if this is an issue with my systems or the launcher.

What do you reckon?
